internal/cmd/keys.go

```diff
@@ -97,15 +97,26 @@ $ infra keys add connector
 				return err
 			}
 
+			var expMsg strings.Builder
+			expMsg.WriteString("This key will expire in ")
+			expMsg.WriteString(ExactDuration(options.TTL))
+			if !resp.Expires.Equal(resp.ExtensionDeadline) {
+				expMsg.WriteString(", and must be used every ")
+				expMsg.WriteString(ExactDuration(options.ExtensionDeadline))
+				expMsg.WriteString(" to remain valid")
+			}
 			cli.Output("Issued access key %q for %q", resp.Name, userName)
+			cli.Output(expMsg.String())
```

Comment:
It would be great to capture this in `TestKeysAddCmd` so that we can see the rendered message. We probably don't even need new test cases, just a new assertion for the output.

We can use the new "golden variable" support in `gotest.tools/v3/assert`. Since we don't have any example of that yet, I put together a commit which adds this test coverage: https://github.com/infrahq/infra/compare/brucemacd/connector-key-feedback...dnephin/demo-golden

Reply:
Thanks for the example, that would have taken me a bit to figure out. I've cherry-picked this commit.